iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/SubnetworkImpl.java

```diff
@@ -686,8 +695,43 @@ public void merge(Network... others) {
 
     @Override
     public Network detach() {
-        //TODO subnetworks API
-        throw new UnsupportedOperationException("Not yet implemented");
+        Set<Identifiable<?>> boundaryElements = getBoundaryElements();
+        checkDetachable(boundaryElements, true);
+
+        long start = System.currentTimeMillis();
+
+        // Remove tie-lines
+        boundaryElements.stream()
```

Comment:
P0 and Q0 of the dangling lines must be updated.

Reply:
This will be done in the subnetworks main PR.